src/SpectatorView.Unity/Assets/SpectatorView/Scripts/SpatialAlignment/SpatialCoordinateSystemManager.cs

```diff
@@ -51,6 +52,55 @@ public class SpatialCoordinateSystemManager : Singleton<SpatialCoordinateSystemM
         private HashSet<INetworkManager> networkManagers = new HashSet<INetworkManager>();
         private ISpatialLocalizationSession currentLocalizationSession = null;
 
+        /// <summary>
+        /// Current Tracking state for the AR/VR Device associated with the application.
+        /// </summary>
+        public TrackingState TrackingState
+        {
+            get
+            {
+                if (!lookedForTrackingObservers)
+                {
+                    List<GameObject> roots = new List<GameObject>();
+                    SceneManager.GetActiveScene().GetRootGameObjects(roots);

```

Comment:
>SceneManager.GetActiveScene().GetRootGameObjects(roots); [](start = 20, length = 56)

I'm not a huge fan of this pattern, would prefer a design where each platform's ITrackingObserver is a component on a child of the SpatialCoordinateSystem inside the SpectatorView prefab, and then referenced here. Each ITrackingObserver could have a "bool IsSupported" that would use platform-specific #if blocks to determine if it should execute on the current platform or not. Then the SpatialCoordinateSystemManager could just find the first assigned supported ITrackingObserver.

Reply:
I'm going to opt to set these up as default prefabs with no override ability that are automatically initialized. That prevents us from having to take a depency on arcore/arkit with our spectator view asmdef. As needed, we can enable customizing these values in the spectator view settings, but right now there isn't much reason to expose more permutations to test.